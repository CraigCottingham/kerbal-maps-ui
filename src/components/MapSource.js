import React from "react"

class MapSource extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      packOptions: this.convertOptions(this.props.packOptions, "pack"),
      pack: this.props.pack,
      onPackChange: this.props.onPackChange,
      bodyOptions: this.convertOptions(this.props.bodyOptions, "body"),
      mapBody: this.props.mapBody,
      onBodyChange: this.props.onBodyChange,
      styleOptions: this.convertOptions(this.props.styleOptions, "style"),
      mapStyle: this.props.mapStyle,
      onStyleChange: this.props.onStyleChange
    }
  }

  setPackOptions(newOptions) {
    this.setState({
      packOptions: this.convertOptions(newOptions, "pack"),
      pack: undefined
    })
  }

  setPack(newPack) {
    this.setState({
      pack: newPack
    })
  }

  setBodyOptions(newOptions) {
    this.setState({
      bodyOptions: this.convertOptions(newOptions, "body"),
      mapBody: undefined
    })
  }

  setMapBody(newBody) {
    this.setState({
      mapBody: newBody
    })
  }

  setStyleOptions(newOptions) {
    this.setState({
      styleOptions: this.convertOptions(newOptions, "style"),
      mapStyle: undefined
    })
  }

  setMapStyle(newStyle) {
    this.setState({
      mapStyle: newStyle
    })
  }

  convertOptions(options, prefix) {
    return options.map((option) =>
      <option key={`${prefix}_` + option.value} value={option.value} disabled={option.disabled ? true : false}>{option.label}</option>
    )
  }

  handleInputChange(event) {
    const target = event.target
    const name = target.name
    const value = target.value

    this.setState({
      [name]: value
    })

    switch (name) {
      case 'pack':
        if (this.state.onPackChange) {
          this.state.onPackChange(value)
        }
        break;

      case 'mapBody':
        if (this.state.onBodyChange) {
          this.state.onBodyChange(value)
        }
        break;

      case 'mapStyle':
        if (this.state.onStyleChange) {
          this.state.onStyleChange(value)
        }
        break;

      default:
        break;
    }
  }

  render() {
    return (
      <form>
        <div className="form-group">
          <label>
            Planet Pack
            <select name="pack" value={this.state.pack} className="form-control" onChange={this.handleInputChange.bind(this)}>
              {this.state.packOptions}
            </select>
          </label>
        </div>
        <div className="form-group">
          <label>
            Body
            <select name="mapBody" value={this.state.mapBody} className="form-control" onChange={this.handleInputChange.bind(this)}>
              {this.state.bodyOptions}
            </select>
          </label>
        </div>
        <div className="form-group">
          <label>
            Style
            <select name="mapStyle" value={this.state.mapStyle} className="form-control" onChange={this.handleInputChange.bind(this)}>
              {this.state.styleOptions}
            </select>
          </label>
        </div>
      </form>
    )
  }
}

export default MapSource
