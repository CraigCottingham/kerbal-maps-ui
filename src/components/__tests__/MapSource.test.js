import React from "react"
import MapSource from "../MapSource"
import renderer from "react-test-renderer"

it("renders correctly", () => {
  const packOptions = [
    {value: "a", label: "A"},
    {value: "b", label: "B"},
    {value: "c", label: "C"}
  ]

  const bodyOptions = [
    {value: "d", label: "D"},
    {value: "e", label: "E"},
    {value: "f", label: "F"}
  ]

  const styleOptions = [
    {value: "g", label: "G"},
    {value: "h", label: "H"},
    {value: "i", label: "I"}
  ]

  const tree = renderer
    .create(
      <MapSource packOptions={packOptions} bodyOptions={bodyOptions} styleOptions={styleOptions} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
