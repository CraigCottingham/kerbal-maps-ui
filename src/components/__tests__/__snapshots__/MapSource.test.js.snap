// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <label>
      Planet Pack
      <select
        className="form-control"
        name="pack"
        onChange={[Function]}
      >
        <option
          disabled={false}
          value="a"
        >
          A
        </option>
        <option
          disabled={false}
          value="b"
        >
          B
        </option>
        <option
          disabled={false}
          value="c"
        >
          C
        </option>
      </select>
    </label>
  </div>
  <div
    className="form-group"
  >
    <label>
      Body
      <select
        className="form-control"
        name="mapBody"
        onChange={[Function]}
      >
        <option
          disabled={false}
          value="d"
        >
          D
        </option>
        <option
          disabled={false}
          value="e"
        >
          E
        </option>
        <option
          disabled={false}
          value="f"
        >
          F
        </option>
      </select>
    </label>
  </div>
  <div
    className="form-group"
  >
    <label>
      Style
      <select
        className="form-control"
        name="mapStyle"
        onChange={[Function]}
      >
        <option
          disabled={false}
          value="g"
        >
          G
        </option>
        <option
          disabled={false}
          value="h"
        >
          H
        </option>
        <option
          disabled={false}
          value="i"
        >
          I
        </option>
      </select>
    </label>
  </div>
</form>
`;
